From: WANG Cong <amwang@redhat.com>
Date: Thu, 17 Dec 2009 15:27:05 -0800
Subject: [stable] kernel/sysctl.c: fix the incomplete part of sysctl_max_map_count-should-be-non-negative.patch
Bugzilla: 555084

(Pulled from -stable series)

commit 3e26120cc7c819c97bc07281ca1fb9017cfe9a39 upstream.

It is a mistake that we used 'proc_dointvec', it should be
'proc_dointvec_minmax', as in the original patch.

Signed-off-by: WANG Cong <amwang@redhat.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

Signed-off-by: Aristeu Rozanski <aris@redhat.com>

diff --git a/kernel/sysctl.c b/kernel/sysctl.c
index e9d896a..dc87fa0 100644
--- a/kernel/sysctl.c
+++ b/kernel/sysctl.c
@@ -1644,7 +1644,7 @@ static struct ctl_table debug_table[] = {
 		.data		= &show_unhandled_signals,
 		.maxlen		= sizeof(int),
 		.mode		= 0644,
-		.proc_handler	= proc_dointvec,
+		.proc_handler	= proc_dointvec_minmax,
 		.extra1		= &zero,
 	},
 #endif
